### PR TITLE
CIV-6051 new event added

### PR DESCRIPTION
**Pull request by maciejmodzelewski**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/CIV-6051

### Change description ###
A new event added

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by hmcts-version1-ignacio — DISMISSED** [review]

File: ccd-definition/AuthorisationCaseEvent/AuthorisationCaseEventCaseProg-nonprod.json [hunk 1]
```diff
@@ -60,5 +60,23 @@
         "CRUD": "R"
       }
     ]
+  },
+  {
+    "CaseTypeID": "CIVIL",
+    "CaseEventID": "TRIAL_READINESS",
+    "AccessControl": [
+      {
+        "UserRoles": [
```
Comment (cj1993) [inline ccd-definition/AuthorisationCaseEvent/AuthorisationCaseEventCaseProg-nonprod.json:69]: Is this only for solicitors ? Judge / LA does not need access ?

File: ccd-definition/CaseEvent/User/UserEventsCaseProg-nonprod.json [hunk 2]
```diff
@@ -67,6 +67,22 @@
     "ShowEventNotes": "N",
     "RetriesTimeoutAboutToStartEvent": 0,
     "RetriesTimeoutURLAboutToSubmitEvent": 0
+  },
+  {
+    "CaseTypeID": "CIVIL",
+    "ID": "TRIAL_READINESS",
+    "Name": "Trial Readiness",
+    "Description": "Trial Readiness",
```
Comment (cj1993) [inline ccd-definition/CaseEvent/User/UserEventsCaseProg-nonprod.json:75]: Is the post & pre state supposed to be the same ?

File: ccd-definition/CaseEvent/User/UserEventsCaseProg-nonprod.json [hunk 3]
```diff
@@ -67,6 +67,22 @@
...
+    "PreConditionState(s)": "PREPARE_FOR_HEARING_CONDUCT_HEARING",
+    "PostConditionState": "PREPARE_FOR_HEARING_CONDUCT_HEARING",
+    "SecurityClassification": "Public",
+    "CallBackURLAboutToStartEvent": "${CCD_DEF_CASE_SERVICE_BASE_URL}/cases/callbacks/about-to-start",
+    "CallBackURLAboutToSubmitEvent": "${CCD_DEF_CASE_SERVICE_BASE_URL}/cases/callbacks/about-to-submit",
+    "CallBackURLSubmittedEvent": "${CCD_DEF_CASE_SERVICE_BASE_URL}/cases/callbacks/submitted",
+    "ShowSummary": "N",
+    "ShowEventNotes": "N",
+    "RetriesTimeoutAboutToStartEvent": 0,
```
Comment (cj1993) [inline ccd-definition/CaseEvent/User/UserEventsCaseProg-nonprod.json:84]: I believe this integrates with Camunda ? If so we will need publish : Y

File: ccd-definition/AuthorisationCaseEvent/AuthorisationCaseEventCaseProg-nonprod.json (hunk 1 above)
Comment (maciejmodzelewski) [reply ccd-definition/AuthorisationCaseEvent/AuthorisationCaseEventCaseProg-nonprod.json:69]: [CIV-6051](https://tools.hmcts.net/jira/browse/CIV-6051) in the user story and acceptance criteria only mentions claimant/defendant (LR)

File: ccd-definition/CaseEvent/User/UserEventsCaseProg-nonprod.json (hunk 2 above)
Comment (maciejmodzelewski) [reply ccd-definition/CaseEvent/User/UserEventsCaseProg-nonprod.json:75]: [CIV-6051](https://tools.hmcts.net/jira/browse/CIV-6051) says:
Event should be available within the following states in [Case Progression - MVP Requirements - State and Event Model](https://tools.hmcts.net/confluence/pages/viewpage.action?pageId=1593580315#CaseProgressionMVPRequirements-StateandEventModel):
PREPARE_FOR_HEARING_CONDUCT_HEARING

I'm assuming that within means that pre and post conditions are the same.

File: ccd-definition/CaseEvent/User/UserEventsCaseProg-nonprod.json (hunk 3 above)
Comment (maciejmodzelewski) [reply ccd-definition/CaseEvent/User/UserEventsCaseProg-nonprod.json:84]: I'm not sure about this. I've done the code the way Max advised me, but I will consult him again on this.

File: ccd-definition/CaseEvent/User/UserEventsCaseProg-nonprod.json (hunk 2 above)
Comment (cj1993) [reply ccd-definition/CaseEvent/User/UserEventsCaseProg-nonprod.json:75]: No, this is the pre state. Speak to Doug about post state requirement

File: ccd-definition/CaseEvent/User/UserEventsCaseProg-nonprod.json (hunk 3 above)
Comment (cj1993) [reply ccd-definition/CaseEvent/User/UserEventsCaseProg-nonprod.json:84]: If this integrates with Camunda (speak to Doug) then you need publish : Y

Comment (m-meulendijks-v1) [reply ccd-definition/CaseEvent/User/UserEventsCaseProg-nonprod.json:84]: I am not aware as of yet of this integrating with camunda.
There are trial-ready notifications and schedulers in camunda, but they do not relate directly to this event - they are just named similarly.

It seems to me to be fine as is - when the designs for Trial Readiness come back, we should be able to ascertain what needs to go into the event, or if anything needs to integrate with camunda.

File: ccd-definition/AuthorisationCaseEvent/AuthorisationCaseEventCaseProg-nonprod.json (hunk 1 above)
Comment (maciejmodzelewski) [reply ccd-definition/AuthorisationCaseEvent/AuthorisationCaseEventCaseProg-nonprod.json:69]: I consulted this with Doug, and he agrees.

File: ccd-definition/CaseEvent/User/UserEventsCaseProg-nonprod.json (hunk 2 above)
Comment (maciejmodzelewski) [reply ccd-definition/CaseEvent/User/UserEventsCaseProg-nonprod.json:75]: I talked to Doug, and he said that this is correct.

File: ccd-definition/CaseEvent/User/UserEventsCaseProg-nonprod.json (hunk 3 above)
Comment (maciejmodzelewski) [reply ccd-definition/CaseEvent/User/UserEventsCaseProg-nonprod.json:84]: I talked to Doug, but he said that he doesn't know about this, so he suggested to follow what Max said.

**Review by cj1993 — DISMISSED** [review]

**Review by nagaraju-deekonda — DISMISSED** [review]

**Review by hmcts-version1-ignacio — APPROVED** [review]

**Review by hmcts-version1-ignacio — APPROVED** [review]

**Review by m-meulendijks-v1 — APPROVED** [review]

**Review by drummondjm — DISMISSED** [review]

**Review by hmcts-version1-ignacio — DISMISSED** [review]

**Review by drummondjm — APPROVED** [review]

**Review by m-meulendijks-v1 — DISMISSED** [review]

**Review by hmcts-version1-ignacio — DISMISSED** [review]

**Review by filippo-possenti-version1 — APPROVED** [review]

**Review by m-meulendijks-v1 — APPROVED** [review]

**Review by drummondjm — APPROVED** [review]

**Pull request merged**